overlord/snapstate/snapmgr.go

```diff
@@ -540,6 +546,99 @@ func (m *SnapManager) ensureForceDevmodeDropsDevmodeFromState() error {
 	return nil
 }
 
+// changeInFlight returns true if there is any change in the state
+// in non-ready state.
+func changeInFlight(st *state.State) bool {
+	for _, chg := range st.Changes() {
+		if !chg.Status().Ready() {
+			// another change already in motion
+			return true
+		}
+	}
+	return false
+}
+
+// ensureSnapdSnapTransition will migrate systems to use the "snapd" snap
+func (m *SnapManager) ensureSnapdSnapTransition() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
```

Comment:
afaict this is missing a check that we are on classic, we will fail anyway in doInstall on core (on core the flag is not enough, we do really need a remodel)